shared/data/core/src/main/java/com/ivy/data/db/migration/Migration129to130_LoanIncludeNote.kt

```diff
@@ -0,0 +1,10 @@
+package com.ivy.data.db.migration
+
+import androidx.room.migration.Migration
+import androidx.sqlite.db.SupportSQLiteDatabase
+
+class Migration129to130_LoanIncludeNote : Migration(129, 130)  {
```

Comment:
Can you add a migration test for this one? See how I've done it for the other migrations

Reply:
@Test
    fun migrate129to130_LoanIncludeNote() {
        // given
        helper.createDatabase(TestDb, 129).apply {
            // Insert some data using SQL queries.
            val insertSql = """
                INSERT INTO loans (name, amount, type, color, icon, orderNum, accountId, isSynced, isDeleted, dateTime, id) 
                VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);
            """.trimIndent()

            val preparedStatement = compileStatement(insertSql).apply {
                // Bind values
                bindString(1, "Loan 1")
                bindDouble(2, 123.50)
                bindString(3, LoanType.BORROW.name) // Assuming you store enum as name
                bindLong(4, 13)
                bindString(5, "ic")
                bindDouble(6, 3.14)
                bindString(7, UUID.randomUUID().toString())
                bindLong(8, 1)
                bindLong(9, 0)
                bindString(10, null)
                bindString(11, UUID.randomUUID().toString())
            }
            preparedStatement.executeInsert()
            close()
        }

        // when
        val newDb = helper.runMigrationsAndValidate(
            TestDb,
            130,
            true,
            Migration129to130_LoanIncludeNote()
        )

        // then
        newDb.query("SELECT * FROM loans").apply {
            moveToFirst() shouldBe true
            getString(0) shouldBe "Loan 1"
            getDouble(1) shouldBe 123.50
            getString(2) shouldBe LoanType.BORROW.name
        }
        newDb.close()
    }

Will this do ? just confirming berfore commit....added this in` IvyRoomDatabaseMigrationTest.kt `

Reply:
> ```
>  @Test
> fun migrate129to130_LoanIncludeNote() {
>     // given
>     helper.createDatabase(TestDb, 129).apply {
>         // Insert some data using SQL queries.
>         val insertSql = """
>             INSERT INTO loans (name, amount, type, color, icon, orderNum, accountId, isSynced, isDeleted, dateTime, id) 
>             VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);
>         """.trimIndent()
> 
>         val preparedStatement = compileStatement(insertSql).apply {
>             // Bind values
>             bindString(1, "Loan 1")
>             bindDouble(2, 123.50)
>             bindString(3, LoanType.BORROW.name) // Assuming you store enum as name
>             bindLong(4, 13)
>             bindString(5, "ic")
>             bindDouble(6, 3.14)
>             bindString(7, UUID.randomUUID().toString())
>             bindLong(8, 1)
>             bindLong(9, 0)
>             bindString(10, "")
>             bindString(11, UUID.randomUUID().toString())
>         }
>         preparedStatement.executeInsert()
>         close()
>     }
> 
>     // when
>     val newDb = helper.runMigrationsAndValidate(
>         TestDb,
>         130,
>         true,
>         Migration129to130_LoanIncludeNote()
>     )
> 
>     // then
>     newDb.query("SELECT * FROM loans").apply {
>         moveToFirst() shouldBe true
>         getString(0) shouldBe "Loan 1"
>         getDouble(1) shouldBe 123.50
>         getString(2) shouldBe LoanType.BORROW.name
>     }
>     newDb.close()
> }
> ```
> 
> Will this do ? just confirming berfore commit....added this in`IvyRoomDatabaseMigrationTest.kt`

take a look at this and let me know i'll commit it

Reply:
Please commit it so I can comment over the code, LGTM in general. Add `moveToNext() should be false` and better extract the migration test in a new file like I did for my last migration